src/coreclr/src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -3146,8 +3146,10 @@ HRESULT ProfToEEInterfaceImpl::GetAppDomainStaticAddress(ClassID classId,
         return CORPROF_E_DATAINCOMPLETE;
     }
 
-    if (typeHandle.GetModule()->GetLoaderAllocator() == NULL ||
-        typeHandle.GetModule()->GetLoaderAllocator()->GetExposedObject() == NULL)
+    // We might have caught a collectible assembly in the middle of being collected
+    Module *pModule = typeHandle.GetModule();
+    if (pModule->IsCollectible() &&
+        (pModule->GetLoaderAllocator() == NULL || pModule->GetLoaderAllocator()->GetExposedObject() == NULL))
```

Comment:
If this is the error we want to keep, can you update the doc comment atop saying "or if the type's assembly has been collected". Also, out of curiosity, is this the address for static fields in general?

Reply:
Sure, I can update the comment. The api exists because profilers don't have another way to inspect static fields. If a class has a static field it isn't part of the object on the heap, so you have to use this api to get a pointer to the field so you can inspect it.